src/api/lastfm.ts

```diff
@@ -0,0 +1,61 @@
+const API_KEY = '120a87ebf9c030970ede045954b05d26';
+const BASE_URL = 'https://ws.audioscrobbler.com/2.0/?format=json';
+
+/**
+ * Получает список самых популярных исполнителей с сервиса Last.fm
+ * @returns {Promise<any[]>} Массив объектов исполнителей
+ */
+export async function getTopArtists() {
+  const res = await fetch(`${BASE_URL}&method=chart.gettopartists&api_key=${API_KEY}`);
+  return (await res.json()).artists.artist;
+}
+
+/**
+ * Получает список самых популярных треков с сервиса Last.fm
+ * @returns {Promise<any[]>} Массив объектов треков
+ */
+export async function getTopTracks() {
+  const res = await fetch(`${BASE_URL}&method=chart.gettoptracks&api_key=${API_KEY}`);
+  return (await res.json()).tracks.track;
+}
+
+/**
+ * Выполняет поиск исполнителей по строке запроса
+ * @param {string} query - Имя или часть имени исполнителя
+ * @returns {Promise<any[]>} Массив подходящих исполнителей
+ */
+export async function searchArtists(query: string) {
+  const res = await fetch(`${BASE_URL}&method=artist.search&artist=${encodeURIComponent(query)}&api_key=${API_KEY}`);
+  return (await res.json()).results.artistmatches.artist;
+}
+
+/**
+ * Выполняет поиск треков по строке запроса
+ * @param {string} query - Название трека
+ * @returns {Promise<any[]>} Массив подходящих треков
+ */
+export async function searchTracks(query: string) {
+  const res = await fetch(`${BASE_URL}&method=track.search&track=${encodeURIComponent(query)}&api_key=${API_KEY}`);
+  return (await res.json()).results.trackmatches.track;
```

Comment:
хорошей привычкой будет использование optional chaining в подобных констуркциях 
https://learn.javascript.ru/optional-chaining


```suggestion
  return (await res.json())?.results?.trackmatches?.track;
```